src/Reflection/ClassReflection.php

```diff
@@ -357,9 +405,60 @@ public function getParents(): array
 	 */
 	public function getInterfaces(): array
 	{
-		return array_map(function (\ReflectionClass $interface): ClassReflection {
-			return $this->broker->getClass($interface->getName());
-		}, $this->getNativeReflection()->getInterfaces());
+		$interfaces = [];
+
+		$parent = $this->getParentClass();
+		if ($parent !== false) {
+			foreach ($parent->getInterfaces() as $interface) {
+				$interfaces[$interface->getName()] = $interface;
+			}
+		}
+
+		$tags = $this->getImplementsTags();
```

Comment:
`$implementsTags`

Reply:
Maybe `@implements` shouldn't be read at all above an interface? Only `@extends`...

Reply:
+1. We will also need a rule to forbid `@implements` on interfaces.